code/src/sixsq/nuvla/server/resources/session.clj

```diff
@@ -267,6 +268,7 @@ status, a 'set-cookie' header, and a 'location' header with the created
   (try
     (let [authn-info (auth/current-authentication request)
           body       (if (u/is-form? headers) (u/convert-form :template form-params) body)
+          _          (-> body :redirect-url config-nuvla/throw-is-not-authorised-redirect-url)
```

Comment:
I see there is a unit test for `group`, but the updated `session` here and `user` below are not tested.

Reply:
The critical redirection is in group resource. user and session redirection are just to return error messages. This is why it must be tested for group. I will add a test for user and more for the authorize-url function